src/redux/sagas/order.saga.js

```diff
@@ -0,0 +1,30 @@
+import axios from 'axios';
+import { put, takeLatest } from 'redux-saga/effects';
+
+function* fetchCustomerOrders() {
+  try {
+    const response = yield axios.get('/api/orders');
+    yield put({
+      type: 'SET_CUSTOMER_ORDERS',
+      payload: response.data
+    })
+  } catch (err) {
+    console.error(err.message);
+  }
+}
+
+function* addOrder(action) {
+  console.log("action payload,", action.payload)
+  try{
+    const response = yield axios.post('/api/orders/shipping', action.payload);
+  } catch(err) {
+    console.error(err.message);
+  }
+}
+
+function* orderSaga() {
+  yield takeLatest('FETCH_CUSTOMER_ORDERS', fetchCustomerOrders);
+  yield takeLatest('ADD_ORDER', addOrder);
```

Comment:
Be careful with Action type naming here. 

We're calling it `ADD_ORDER`, but we're actually a `PUT` to the `/orders/shipping` endpoint -- those don't match in my mind, and may cause confusion when we also need to do a `POST /initialOrder` endpoint.


Related to my previous comment, this is a good reason for using REST API conventions, as we can follow through on those conventions in our client side:

- `GET /orders` ➡️  `FETCH_ORDERS`
- `POST /orders` ➡️  `CREATE_ORDER`
- `PUT /orders/:id` ➡️  `UPDATE_ORDER`

When all of our different resource types follow that same `GET/POST/PUT`, ` FETCH/CREATE/UPDATE` naming convention, it makes it's easier to quickly look at the code, and understand what it's up to. 🙂